### ISSUE TITLE
Allow read_survey() to specify col_types

### ISSUE THREAD

**Issue by jntrcs** [ropensci/qualtRics#148]
Peeking at the source code, it appears read_survey does some manipulation and then uses read_csv() to parse the file. However, it relies on a default handling of col_types. It would be really convenient if we could use the col_types argument of read_csv() to specify the col_types.

In my use case, read_csv is incorrectly guessing that it's a numeric column when I would rather have it as a character. An example row is "1,2323,323,3455,45,2343" and it's stripping out the commas and trying to convert it to a number, but due to the size of the numbers we're losing digits to rounding. 

I think this could potentially be done with the addition of ... that is passed onto read_csv. If someone is interested in giving me some guidance I would attempt a pull request.

**Comment by jmobrien** [comment]
The line where that all happens is in `read_survey`, specifically a call to `readr::type_convert`.  This could take a `col_types` argument to do what you're discussing (see help for `type_convert` and `cols`:

```r
# This already here:
  rawdata <- readr::type_convert(rawdata,
                                 locale = readr::locale(tz = time_zone),
# This gets added:
                                 col_types = cols([var] = ["c"/col_character()]
)
```
You'd need to be able to pass an arbitrary number of items from `fetch_survey` to that `cols()` and format it correctly.  Shouldn't be too hard, but you might refer to @juliasilge for views about how that should be implemented in a user-friendly way consistent with the current UI philosophy.  

Plus, ideally, some prior checking to ensure the user actually passed correct variable names, maybe some checking for duplicated names.  Though you don't actually *have* those names until the content is downloaded, so that might take some thought.

Also, @juliasilge, while we're at it, maybe some columns should be type-fixed by default?  I'm thinking of the standard ExternalDataReference column, which lists the data that authenticator blocks check against, and should probably be fixed to character.  I say this b/c my team experienced some issues with numeric ID codes when some of them  had leading zeros that got dropped when those codes were interpreted as standard numbers (This actually happened with Excel and csv's prior to uploading them in our Q library, thereby messing up some our user authentication.  Fun times.  But the same principle follows when downloading, and could lead to mismatches on that important column.)

**Comment by juliasilge** [comment]
This could be a great addition to this package, and I would love to see a PR. The ideas that @jmobrien outlines here have some helpful thoughts.

One note is that we need to include tests with any new PR. Unfortunately we are in an awkward place right now with the testing suite because it currently has very old tests written with an outdated framework. Moving forward, I want to use vcr and probably webmockr for tests, but I haven't gotten a chance to switch all the old tests over yet. New tests *can* be written this way, though; rOpenSci has a [great resource here on how to get started](https://books.ropensci.org/http-testing/).

(As a side note, I am definitely looking for help on switching over and building out the existing testing suite.)

**Comment by jntrcs** [comment]
So, sorry I left this hanging for so long, but I am interested in working on it.

The solution I would propose to this would be pretty minimalistic. Why don't we just add the col_types() argument as an argument to read_survey (and fetch_survey) with null default, and pass the argument unchanged to readr::type_convert(). Then our documentation can mention what the argument is for (to fix incorrectly guessed column types) and point people to readr's documentation for the formatting and usage. 

Since none of these changes would have any relation to the API calls (it only involves the processing of the raw data file), I don't think any tests related to them would need to invoke the vcr or wemockr packages.

**Comment by juliasilge** [comment]
Closed in fecbb88331de340f050138dd666ed226cfce4d4c thanks to @jntrcs 🙌
